src/photos/targets/services/onPhotoUpload.js

```diff
@@ -168,27 +142,40 @@ const recomputeParameters = async setting => {
   }
   log('info', `Compute clustering parameters on ${files.length} photos`)
 
-  const dataset = prepareDataset(files)
+  const dataset = await prepareDataset(files)
   const epsTemporal = computeEpsTemporal(dataset, PERCENTILE)
   const epsSpatial = computeEpsSpatial(dataset, PERCENTILE)
   return createParameter(dataset, epsTemporal, epsSpatial)
 }
 
-const onPhotoUpload = async () => {
-  log('info', `Service called with COZY_URL: ${process.env.COZY_URL}`)
-
-  let setting = await readSetting()
-  const lastSeq = setting ? setting.lastSeq : 0
-
-  const changes = await getChanges(lastSeq)
+const runClustering = async (setting, since, filteredIds) => {
+  const changes = await getChanges(since, CHANGES_RUN_LIMIT, filteredIds)
   if (!changes || changes.photos.length < 1) {
```

Comment:
I don't think `changes ` can return nothing, so the condition could be simplified, no?